File: forge/src/main/java/net/shiroha233/roadweaver/network/forge/ClientNetBridgeImpl.java
package net.shiroha233.roadweaver.network.forge;

import net.minecraft.resources.ResourceLocation;
import net.shiroha233.roadweaver.client.map.MapLoadPhase;

/**
 * Forge 平台网络桥接实现
 */
public final class ClientNetBridgeImpl {
    private ClientNetBridgeImpl() {}

    public static void requestSnapshot(int requestSeq,
                                       ResourceLocation dimensionId,
                                       MapLoadPhase phase,
                                       int responseIndex,
                                       int minX,
                                       int minZ,
                                       int maxX,
                                       int maxZ) {
        MapNetworkForge.requestSnapshot(requestSeq, dimensionId, phase, responseIndex, minX, minZ, maxX, maxZ);
    }

    public static void requestTeleport(int x, int y, int z) {
        MapNetworkForge.requestTeleport(x, y, z);
    }

    public static void requestManualConnect(int ax, int az, int bx, int bz) {
        MapNetworkForge.requestManualConnect(ax, az, bx, bz);
    }
}

File: forge/src/main/kotlin/net/shiroha233/roadweaver/network/forge/ClientNetBridgeImpl.kt
package net.shiroha233.roadweaver.network.forge

@Suppress("MemberVisibilityCanBePrivate")
object ClientNetBridgeImpl {
    @JvmStatic
    fun requestSnapshot(minX: Int, minZ: Int, maxX: Int, maxZ: Int) {
        MapNetworkForge.requestSnapshot(minX, minZ, maxX, maxZ)
    }

    @JvmStatic
    fun requestTeleport(x: Int, y: Int, z: Int) {
        MapNetworkForge.requestTeleport(x, y, z)
    }

    @JvmStatic
    fun requestManualConnect(ax: Int, az: Int, bx: Int, bz: Int) {
        MapNetworkForge.requestManualConnect(ax, az, bx, bz)
    }
}
